pkg/protecode/protecode.go

```diff
@@ -363,10 +381,20 @@ func (pc *Protecode) PollForResult(productID int, timeOutInMinutes string) Resul
 		}
 	}
 
-	if len(response.Result.Components) == 0 || response.Result.Status == statusBusy {
+	if protecodeStillWorking(response.Result.Status) {
 		response, err = pc.pullResult(productID)
-		if err != nil || len(response.Result.Components) == 0 || response.Result.Status == statusBusy {
-			pc.logger.Fatal("No result after polling")
+
+		if len(response.Result.Components) < 1 {
+			// 2020-04-20 d :
+			// We are required to scan all images including 3rd party ones.
+			// We have found that Crossplane makes use docker images that contain no
+			// executable code.
+			// So we can no longer treat an empty Components list as an error.
+			pc.logger.Warn("Protecode scan did not identify any components.")
+		}
+
+		if err != nil || response.Result.Status == statusBusy {
+			pc.logger.Fatalf("No result after polling err: %v protecode status: %v", err, response.Result.Status)
```

Comment:
What if the `err` is `nil`? 
I think we should split this and have dedicated error messages.